### PR TITLE
Run Modernizer

### PR DESCRIPTION
**Pull request by daniel-j-h**
Does what it says.

We probably should add a Travis build for this with the latest Clang, but oh well..

### CODE REVIEW

**Comment by daniel-j-h** [comment]
Only the Travis GCC 4.8 Debug build fails on this. How is it even possible..? The changes look good.
Notably, some cucumber tests fail, with the note that osrm isn't running.

File: datastore.cpp [hunk 1]
```diff
@@ -229,7 +229,7 @@ int main(const int argc, const char *argv[])
         // Allocate a memory layout in shared memory, deallocate previous
         SharedMemory *layout_memory =
             SharedMemoryFactory::Get(layout_region, sizeof(SharedDataLayout));
-        SharedDataLayout *shared_layout_ptr = new (layout_memory->Ptr()) SharedDataLayout();
```
Comment (TheMarex) [inline datastore.cpp]: Uhm shouldn't this be `auto*`?

Comment (daniel-j-h) [inline datastore.cpp]: Nope, `auto` type deduction works like template type deduction. And `T` vs `T*` matters only for human readability. Here's an example with a nice trick to force the compiler to show me the infered types. In both cases they are `char*`: http://coliru.stacked-crooked.com/a/302af5710763fb14

**Comment by TheMarex** [comment]
Other than the failing travis build, this should be good. Restarted, looks like a blip. :+1:

**Comment by c0nk** [comment]
What about trying to revert the changes one-by-one until it passes?

**Pull request merged**